client/blocks/product-plan-overlap-notices/index.jsx

```diff
@@ -0,0 +1,158 @@
+import { PRODUCT_SHORT_NAMES } from 'lib/products-values/constants';
+
+class ProductPlanOverlapNotices extends Component {
+	static propTypes = {
+		plans: PropTypes.arrayOf( PropTypes.string ).isRequired,
+		products: PropTypes.arrayOf( PropTypes.string ).isRequired,
+		siteId: PropTypes.number,
+
+		// Connected props
+		availableProducts: PropTypes.object,
+		currentPlanSlug: PropTypes.string,
+		purchases: PropTypes.array,
+		selectedSiteId: PropTypes.number,
+
+		// From localize() HoC
+		translate: PropTypes.func.isRequired,
+	};
+
+	hasOverlap() {
+		const { availableProducts, currentPlanSlug, plans, products, purchases } = this.props;
+
+		if ( ! currentPlanSlug || ! purchases || ! availableProducts ) {
+			return false;
+		}
+
+		// Is the current plan among the plans we're interested in?
+		if ( ! includes( plans, currentPlanSlug ) ) {
+			return false;
+		}
+
+		// Is the current product among the products we're interested in?
+		if ( ! this.getCurrentProductSlug() ) {
+			return false;
+		}
+
+		// Does the current plan include the product as a feature?
+		return some( products, productSlug => planHasFeature( currentPlanSlug, productSlug ) );
+	}
+
+	getCurrentProductSlug() {
+		const { products, purchases } = this.props;
+
+		const currentProduct = find( purchases, purchase =>
+			includes( products, purchase.productSlug )
+		);
+		if ( ! currentProduct ) {
+			return null;
+		}
+
+		return currentProduct.productSlug;
+	}
+
+	getCurrentProductName() {
+		const { availableProducts } = this.props;
+		const currentProductSlug = this.getCurrentProductSlug();
+
+		if ( ! currentProductSlug || ! availableProducts[ currentProductSlug ] ) {
+			return '';
+		}
+
+		return availableProducts[ currentProductSlug ].product_name;
+	}
+
+	getCurrentPlanName() {
+		const { availableProducts, currentPlanSlug } = this.props;
+
+		if ( ! availableProducts[ currentPlanSlug ] ) {
+			return '';
+		}
+
+		return availableProducts[ currentPlanSlug ].product_name;
+	}
+
+	getOverlappingFeatureName() {
+		const { availableProducts } = this.props;
+
+		if ( ! this.hasOverlap() ) {
+			return null;
+		}
+
+		const currentProductSlug = this.getCurrentProductSlug();
+		if ( ! currentProductSlug ) {
+			return null;
+		}
+
+		if ( PRODUCT_SHORT_NAMES[ currentProductSlug ] ) {
+			return PRODUCT_SHORT_NAMES[ currentProductSlug ].toLowerCase();
+		}
+
+		if ( availableProducts[ currentProductSlug ] ) {
+			return availableProducts[ currentProductSlug ].product_name;
+		}
+
+		return null;
+	}
+
+	render() {
+		const { selectedSiteId, translate } = this.props;
+
+		return (
+			<Fragment>
+				<QuerySitePlans siteId={ selectedSiteId } />
+				<QuerySitePurchases siteId={ selectedSiteId } />
+				<QueryProductsList />
+
+				{ this.hasOverlap() && (
+					<Notice
+						status="is-warning"
+						text={ translate(
+							'Your %(planName)s Plan includes %(featureName)s. ' +
+								'Looks like you also purchased the %(productName)s product. ' +
```

Comment:
Are we calling it solution now?

Reply:
Maybe 😉 I just used the copy from the latest Figma for this. cc @crunnells and @eeeeevon13 if we should alter `product` to `solution` here.